### PR TITLE
Make SummayLength configurable

### PR DESCRIPTION
**Pull request by Shuenhoy**

### CODE REVIEW

**Comment by Shuenhoy** [comment]
It seems that my code is out of date

File: helpers/content.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ import (
 )
 
 // SummaryLength is the length of the summary that Hugo extracts from a content.
-var SummaryLength = 70
+var SummaryLength = 200
```
Comment (bep) [inline helpers/content.go]: Don't change the default value. That will break many, many sites.

File: hugolib/config.go [hunk 2]
```diff
@@ -132,4 +133,5 @@ func loadDefaultSettingsFor(v *viper.Viper) {
 	v.SetDefault("enableGitInfo", false)
 	v.SetDefault("ignoreFiles", make([]string, 0))
 	v.SetDefault("disableAliases", false)
+	v.SetDefault("SummaryLength", 200)
```
Comment (bep) [inline hugolib/config.go]: Follow the pattern ans start with lower char: SummaryLength > summaryLength

File: helpers/content.go (hunk 1 above)
Comment (bep) [inline helpers/content.go]: As this is moved to a config variable, this variable should be removed.

File: hugolib/config.go (hunk 2 above)
Comment (bep) [inline hugolib/config.go]: As noted above: The default above should remain at 70 (changing that will break many many sites).

File: hugolib/page.go [hunk 3]
```diff
@@ -611,9 +611,9 @@ func (p *Page) setAutoSummary() error {
 	var summary string
 	var truncated bool
 	if p.isCJKLanguage {
-		summary, truncated = helpers.TruncateWordsByRune(p.PlainWords(), helpers.SummaryLength)
+		summary, truncated = helpers.TruncateWordsByRune(p.PlainWords() , p.s.Cfg.GetInt("SummaryLength"))
```
Comment (bep) [inline hugolib/page.go]: summaryLength

**Comment by bep** [comment]
And this needs a line somewhere in the documentation (just add it to this PR).

**Comment by moorereason** [comment]
Replaced by #3924

**Pull request closed without merging**